loader/src/sources/heb/index.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Get official apointment data from H-E-B. This works via a file they
+ * publish to their own website for appointment finders and other consumers like us, although
+ * it does not appear to be officially documented anywhere. It could disappear
+ * in the future without notice.
+ *
+ * Most of the logic and branding information here is based on an original
+ * implementation by Nick Muerdter of VaccineSpotter.org (which has now been
+ * shut down).
+ * https://github.com/GUI/covid-vaccine-spotter/tree/main/src/providers
+ */
+
+const Sentry = require("@sentry/node");
+const { DateTime } = require("luxon");
+const { httpClient } = require("../../utils");
+const { LocationType, VaccineProduct, Available } = require("../../model");
+
+const API_URL =
+  "https://heb-ecom-covid-vaccine.hebdigital-prd.com/vaccine_locations.json";
+
+// Maps H-E-B product names to our product names.
+const PRODUCT_NAMES = {
+  pfizer: VaccineProduct.pfizer,
+  moderna: VaccineProduct.moderna,
+  janssen: VaccineProduct.janssen,
+  pediatric_pfizer: VaccineProduct.pfizerAge5_11,
+};
+
+async function fetchRawData() {
+  const response = await httpClient(API_URL, {
+    // Bust caches with a random querystring.
+    searchParams: { v: Math.random() * 999999999999 },
+    responseType: "json",
+    timeout: 30000,
+  });
+
+  const lastModified = response.headers["last-modified"];
+
+  return {
+    validAt: lastModified
+      ? DateTime.fromHTTP(lastModified, { zone: "utc" }).toISO()
+      : undefined,
+    data: response.body,
+  };
+}
+
+/**
+ * H-E-B has several locations missing a store number and/or
+ * missing a booking URL. Not including those locations in results.
+ * @param {*} states
+ * @returns
+ */
+async function getData(states) {
+  const { validAt, data } = await fetchRawData();
+  const checkedAt = new Date().toISOString();
+
+  return data.locations
+    .filter((location) => Boolean(location.storeNumber))
+    .filter((location) => location.url != null)
```

Comment:
Agreed, will do

Reply:
Oh! It turns out that this result set has a null url when no appointment slots are available. Updating this filter to explicitly filter out stores with no availability. 

Also added the fallback url handling in case we do ever encounter a store with appointment availability but no booking url.

Reply:
> Updating this filter to explicitly filter out stores with no availability.

So the thing here is that, if a location goes from some availability one one run to no availability on the next, we’ll be left with an out-of-date record saying there *is* availability. We still need to send info if availability is none.

Reply:
I see. This definitely highlights the challenges of jumping right in without a lot of context. I had inferred from the productSchema that we only returned results that had availability, since returning a location that has no available products fails the test schema check. It sounds like I should be returning locations with no product availability in the list of results...and should update that `productsSchema` to accept empty arrays since some locations will have no available products. Sound right?

Reply:
> returning a location that has no available products fails the test schema check

Hmmmm, it *shouldn’t.* Plenty of our sources don’t return products. I don’t think `products` is ever marked as required in any of the schemas.

Reply:
> and should update that productsSchema to accept empty arrays since some locations will have no available products

The practice we’ve followed in the API so far is that if there’s no value, you should explicitly send `null` or `undefined`, never an empty array. e.g: https://github.com/usdigitalresponse/univaf/blob/38af404bc2f461e1bb25214229ecc9796d139ae0/loader/src/sources/cdc/api.js#L417-L424

Reply:
:+1: I realized this must be the case an hour after writing this question. Makes sense, updated to do that here as well.